fabrics.c

```diff
@@ -1044,10 +1041,12 @@ int build_options(char *argstr, int max_len, bool discover)
 	    (strncmp(fabrics_cfg.transport, "loop", 4) &&
 	     add_int_argument(&argstr, &max_len, "ctrl_loss_tmo",
 			      fabrics_cfg.ctrl_loss_tmo, true)) ||
+	    (fabrics_cfg.fast_io_fail_tmo != -1 &&
 	    add_int_argument(&argstr, &max_len, "fast_io_fail_tmo",
-				fabrics_cfg.fast_io_fail_tmo, true) ||
+				fabrics_cfg.fast_io_fail_tmo, true)) ||
+	    (fabrics_cfg.tos != -1 &&
```

Comment:
Not sure about this either. So I see that RDMA and TCP are only considering a value >=0, though a comment says

`fabrics.c:573:  opts->tos = -1; /* < 0 == use transport default */  `

From this I'd say -1 is a valid value though the current kernel code filters it out.

Reply:
if you pass tos=-1 to the kernel the latter will print an error and close the connection

Reply:
Yeah, but things don't seem to add up if I read it correctly. tos is initialized with -1 and when user space doesn't set it, the tos value will be ignored. But when we set it via userspace we can only set it in the range of [0,255] and hence we can't disable it later. So my point/question is should -1 not be allowed by the kernel as 'off' value?

Reply:
Ok, got it, will investigate this too